src/extension/src/ActionHandler.py

```diff
@@ -63,9 +66,41 @@ def update(self):
          4. install (if updateMode is UpdateWithInstall)
          5. enable on the new version
          on uninstall the agent deletes removes configuration files"""
-        # todo: in the test run verify if CoreState.json, ExtState.json and the .status files are deleted, if yes, move them to a separate location
-        self.logger.log("Extension updated")
-        return Constants.ExitCode.Okay
+
+        # config folder path is usually something like: /var/lib/waagent/Microsoft.CPlat.Core.LinuxPatchExtension-<version>/config
+        try:
+            self.logger.log("Extension is being updated to the latest version. Copying the required extension artifacts from previous version to the current one")
+            new_version_config_folder = self.ext_env_handler.config_folder
+            extension_pardir = os.path.abspath(os.path.join(new_version_config_folder, os.path.pardir, os.path.pardir))
+            paths_to_all_versions = self.get_all_versions(extension_pardir)
```

Comment:
Log what was calculated to be the parent dir and the list of all versions found on the system.